src/elements/content-uploader/UploadState.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import { useIntl, FormattedMessage } from 'react-intl';
+import { HatWand } from '@box/blueprint-web-assets/illustrations/Medium';
+
+import UploadEmptyState from '../../icons/states/UploadEmptyState';
+import UploadSuccessState from '../../icons/states/UploadSuccessState';
+import UploadStateContent from './UploadStateContent';
+import type { View } from '../../common/types/core';
+
+import { VIEW_ERROR, VIEW_UPLOAD_EMPTY, VIEW_UPLOAD_IN_PROGRESS, VIEW_UPLOAD_SUCCESS } from '../../constants';
+
+import messages from '../common/messages';
+
+import './UploadState.scss';
+
+export interface UploadStateProps {
+    canDrop: boolean;
+    hasItems: boolean;
+    isFolderUploadEnabled: boolean;
+    isOver: boolean;
+    isTouch: boolean;
+    onSelect: () => void;
+    view: View;
+}
+
+const UploadState = ({
+    canDrop,
+    hasItems,
+    isOver,
+    isTouch,
+    view,
+    onSelect,
+    isFolderUploadEnabled,
+}: UploadStateProps) => {
+    const intl = useIntl();
+    let icon;
+    let content;
+    switch (view) {
+        case VIEW_ERROR:
+            icon = <HatWand aria-label={intl.formatMessage(messages.uploadErrorState)} height={126} width={130} />;
+            content = <UploadStateContent message={<FormattedMessage {...messages.uploadError} />} />;
+            break;
+        case VIEW_UPLOAD_EMPTY:
+            icon = <UploadEmptyState title={<FormattedMessage {...messages.uploadEmptyState} />} />;
```

Comment:
nit: i think it's cleaner now to convert these (here and below) `<FormattedMessage />` to `formatMessage` to you don't need both in the file

`title` should be a string but `<FormattedMessage />` is likely returning JSX